src/aligners/diagonal_transition.rs

```diff
@@ -115,6 +116,145 @@
+                            new_cost = s - self.cm.affine[layer_idx].open as usize;
+                            cigar.push(CigarOp::AffineOpen(layer_idx));
+                            layer = None;
+                        }
+                    } else {
+                        let mut command = CigarOp::Match;
+                        // Affine layers
+                        for layer_idx in 0..N {
+                            // Gap close
+                            if fronts[s].affine_mut(layer_idx)[d] > f {
+                                layer = Some(layer_idx);
+                                command = CigarOp::AffineClose(layer_idx);
+                            }
+                            f = max(f, fronts[s].affine_mut(layer_idx)[d]);
+                        }
+                        // Substitution
+                        if let Some(cost) = self.cm.sub {
+                            if fronts[s - cost as usize].m()[d] + 2 > f {
+                                new_cost = s - cost as usize;
+                                layer = None;
+                                command = CigarOp::Mismatch;
+                            }
+                            f = max(f, fronts[s - cost as usize].m()[d] + 2);
+                        }
+                        // Insertion
+                        if let Some(cost) = self.cm.ins {
+                            if fronts[s - cost as usize].m()[d + 1] + 1 > f {
+                                new_cost = s - cost as usize;
+                                layer = None;
+                                new_d = d + 1;
+                                flag = 1;
+                                command = CigarOp::Insertion;
+                            }
+                            f = max(f, fronts[s - cost as usize].m()[d + 1] + 1);
+                        }
+                        // Deletion
+                        if let Some(cost) = self.cm.del {
+                            if fronts[s - cost as usize].m()[d - 1] + 1 > f {
+                                new_cost = s - cost as usize;
+                                layer = None;
+                                new_d = d - 1;
+                                flag = 2;
+                                command = CigarOp::Deletion;
+                            }
+                            f = max(f, fronts[s - cost as usize].m()[d - 1] + 1);
+                        }
+                        cigar.match_push((fronts[s].m()[d] - f) as usize / 2);
+                        cigar.push(command);
+                    }
+                }
+                GapClose => {
+                    todo!();
+                }
+            }
+
+            if flag == 1 {
+                save((f + new_d) as usize / 2 + 1, (f - new_d) as usize / 2);
+            } else if flag == 2 {
+                save((f + new_d) as usize / 2, (f - new_d) as usize / 2 + 1);
+            }
+
+            d = new_d;
+            s = new_cost;
```

Comment:
See if you really need the old values at all. I don't think you do, so you can
just increment `d` and `s` in-place directly.